File: source/kCura.Relativity.Export/Exporters/LoadFileCellFormatters/HtmlCellFormatter.vb
Imports kCura.Relativity.Export.Exports
Imports kCura.Relativity.Export.FileObjects
Imports Relativity

Namespace kCura.Relativity.Export.Exports

	Public Class HtmlCellFormatter
		Implements ILoadFileCellFormatter
		Private _settings As ExportFile
		Private Const ROW_PREFIX As String = "<tr>"
		Private Const ROW_SUFFIX As String = "</tr>"
		Public Sub New(ByVal settings As ExportFile)
			_settings = settings
		End Sub

		Public Function TransformToCell(ByVal contents As String) As String Implements ILoadFileCellFormatter.TransformToCell
			contents = System.Web.HttpUtility.HtmlEncode(contents)
			Return String.Format("{0}{1}{2}", "<td>", contents, "</td>")
		End Function

		Private Function GetNativeHtmlString(ByVal artifact As ObjectExportInfo, ByVal location As String) As String
			If _settings.ArtifactTypeID = ArtifactType.Document AndAlso artifact.NativeCount = 0 Then Return ""
			If Not _settings.ArtifactTypeID = ArtifactType.Document AndAlso Not artifact.FileID > 0 Then Return ""
			Dim retval As New System.Text.StringBuilder
			retval.AppendFormat("<a style='display:block' href='{0}'>{1}</a>", location, artifact.NativeFileName(_settings.AppendOriginalFileName))
			Return retval.ToString
		End Function

		Public ReadOnly Property RowPrefix() As String Implements ILoadFileCellFormatter.RowPrefix
			Get
				Return ROW_PREFIX
			End Get
		End Property

		Public ReadOnly Property RowSuffix() As String Implements ILoadFileCellFormatter.RowSuffix
			Get
				Return ROW_SUFFIX
			End Get
		End Property

		Private Function GetImagesHtmlString(ByVal artifact As ObjectExportInfo) As String
			If artifact.Images.Count = 0 Then Return ""
			Dim retval As New System.Text.StringBuilder
			For Each image As ImageExportInfo In artifact.Images
				Dim loc As String = image.TempLocation
				If Not _settings.VolumeInfo.CopyFilesFromRepository Then
					loc = image.SourceLocation
				End If
				retval.AppendFormat("<a style='display:block' href='{0}'>{1}</a>", loc, image.FileName)
				If _settings.TypeOfImage = ExportFile.ImageType.MultiPageTiff OrElse _settings.TypeOfImage = ExportFile.ImageType.Pdf Then Exit For
			Next
			Return retval.ToString
		End Function

		Public Function CreateImageCell(ByVal artifact As ObjectExportInfo) As String Implements ILoadFileCellFormatter.CreateImageCell
			If Not _settings.ExportImages OrElse _settings.ArtifactTypeID <> ArtifactType.Document Then Return String.Empty
			Return String.Format("<td>{0}</td>", Me.GetImagesHtmlString(artifact))
		End Function

		Public Function CreateNativeCell(ByVal location As String, ByVal artifact As ObjectExportInfo) As String Implements ILoadFileCellFormatter.CreateNativeCell
			Return String.Format("<td>{0}</td>", Me.GetNativeHtmlString(artifact, location))
		End Function
	End Class
End Namespace

File: source/kCura.Relativity.Export.ConvertedToC#/Exporters/LoadFileCellFormatters/HtmlCellFormatter.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Linq;
using kCura.Relativity.Export.Exports;
using kCura.Relativity.Export.FileObjects;
using Relativity;

namespace kCura.Relativity.Export.Exports
{

	public class HtmlCellFormatter : ILoadFileCellFormatter
	{
		private ExportFile _settings;
		private const string ROW_PREFIX = "<tr>";
		private const string ROW_SUFFIX = "</tr>";
		public HtmlCellFormatter(ExportFile settings)
		{
			_settings = settings;
		}

		public string TransformToCell(string contents)
		{
			contents = System.Web.HttpUtility.HtmlEncode(contents);
			return string.Format("{0}{1}{2}", "<td>", contents, "</td>");
		}

		private string GetNativeHtmlString(ObjectExportInfo artifact, string location)
		{
			if (_settings.ArtifactTypeID == (int)ArtifactType.Document && artifact.NativeCount == 0)
				return "";
			if (_settings.ArtifactTypeID != (int)ArtifactType.Document && !(artifact.FileID > 0))
				return "";
			System.Text.StringBuilder retval = new System.Text.StringBuilder();
			retval.AppendFormat("<a style='display:block' href='{0}'>{1}</a>", location, artifact.NativeFileName(_settings.AppendOriginalFileName));
			return retval.ToString();
		}

		public string RowPrefix {
			get { return ROW_PREFIX; }
		}

		public string RowSuffix {
			get { return ROW_SUFFIX; }
		}

		private string GetImagesHtmlString(ObjectExportInfo artifact)
		{
			if (artifact.Images.Count == 0)
				return "";
			System.Text.StringBuilder retval = new System.Text.StringBuilder();
			foreach (ImageExportInfo image in artifact.Images) {
				string loc = image.TempLocation;
				if (!_settings.VolumeInfo.CopyFilesFromRepository) {
					loc = image.SourceLocation;
				}
				retval.AppendFormat("<a style='display:block' href='{0}'>{1}</a>", loc, image.FileName);
				if (_settings.TypeOfImage == ExportFile.ImageType.MultiPageTiff || _settings.TypeOfImage == ExportFile.ImageType.Pdf)
					break; // TODO: might not be correct. Was : Exit For
			}
			return retval.ToString();
		}

		public string CreateImageCell(ObjectExportInfo artifact)
		{
			if (!_settings.ExportImages || _settings.ArtifactTypeID != (int)ArtifactType.Document)
				return string.Empty;
			return string.Format("<td>{0}</td>", this.GetImagesHtmlString(artifact));
		}

		public string CreateNativeCell(string location, ObjectExportInfo artifact)
		{
			return string.Format("<td>{0}</td>", this.GetNativeHtmlString(artifact, location));
		}
	}
}

